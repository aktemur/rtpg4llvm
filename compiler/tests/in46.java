/*compunit*/
void foo(int *x[], int y[], int z[]){
    int rows[3];
    int rows[3] = 4;
}
void bar(int x, int y[]){
    int *rows[5];
    int *rows[5] = 3;
}
