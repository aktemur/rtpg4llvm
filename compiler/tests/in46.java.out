mkProgram(2, new FuncDeclaration(new VoidType(),
      "foo",
      mkFormals(3, new FuncParam(new ArrayType(new PtrType(new IntType())), "x"),
         new FuncParam(new ArrayType(new IntType()), "y"),
         new FuncParam(new ArrayType(new IntType()), "z")),
      mkScopedStmt(2, new VarDecl(new ArrayType(new IntType(),3), "rows"),
      new VarDecl(new ArrayType(new IntType(),3), "rows", mkConst(4)))),
   new FuncDeclaration(new VoidType(),
      "bar",
      mkFormals(2, new FuncParam(new IntType(), "x"),
         new FuncParam(new ArrayType(new IntType()), "y")),
      mkScopedStmt(2, new VarDecl(new ArrayType(new PtrType(new IntType()),5), "rows"),
      new VarDecl(new ArrayType(new PtrType(new IntType()),5), "rows", mkConst(3)))))
